source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -191,6 +191,7 @@ void Filter::onDestroy() {
     state_ = State::Complete;
     client_->cancel();
   } else if (client_ != nullptr) {
+    ASSERT(request_headers_ != nullptr);
```

Comment:
What will happen if the even the decodeHeaders wasn't be called and then the the onDestory() is called. For example, a filter before ratelimit to send a local reply when processing the request headers?